src/tabs/tab-header-bar.tsx

```diff
@@ -332,12 +348,55 @@ const TabTrigger = forwardRef(
     ref: React.Ref<HTMLElement>
   ) => {
     const refObject = useRef<HTMLElement>(null);
+    const childRefObject = useRef<HTMLElement>(null);
     const mergedRef = useMergeRefs(refObject, ref);
     const { tabIndex } = useSingleTabStopNavigation(refObject);
+    const isDisabledWithReason = tab.disabled && !!tab.disabledReason;
+    const [showTooltip, setShowTooltip] = useState(false);
+    const { targetProps, descriptionEl } = useHiddenDescription(tab.disabledReason);
+    const children = (
+      <>
+        <span className={styles['tabs-tab-label']} ref={childRefObject}>
+          {tab.label}
+        </span>
+        {isDisabledWithReason && (
+          <>
+            {descriptionEl}
+            {showTooltip && (
+              <Tooltip
+                className={styles['disabled-reason-tooltip']}
+                trackRef={childRefObject}
+                value={tab.disabledReason!}
+              />
+            )}
+          </>
+        )}
+      </>
+    );
+
+    const handlers = {
+      onFocus: isDisabledWithReason ? () => setShowTooltip(true) : undefined,
+      onBlur: isDisabledWithReason ? () => setShowTooltip(false) : undefined,
+      onMouseEnter: isDisabledWithReason ? () => setShowTooltip(true) : undefined,
+      onMouseLeave: isDisabledWithReason ? () => setShowTooltip(false) : undefined,
+    };
+
+    const commonProps = {
+      ...elementProps,
+      ...(isDisabledWithReason ? targetProps : {}),
```

Comment:
Can we add handlers the same way for consistency? Like:

```
const commonProps = {
  ...elementProps,
  ...(isDisabledWithReason ? targetProps : {}),
  ...(isDisabledWithReason ? handlers : {}),
  ref: mergedRef,
  tabIndex
};
```

Reply:
updated